components/styles/styler.component.ts

```diff
@@ -421,36 +432,95 @@ export class HsStylerComponent {
       for (const subStyle of resolvedStyle) {
         this.parseStyle(subStyle);
       }
-    } else if (this.HsUtilsService.instOf(resolvedStyle, Style)) {
+    } else {
       this.parseStyle(resolvedStyle);
     }
   }
 
-  private parseStyle(subStyle: Style) {
-    if (subStyle.getStroke()?.getColor()) {
+  private parseStyle(style: Style) {
+    const subStyle: Style = this.HsSaveMapService.serializeStyle(style);
+    if (subStyle.stroke?.color) {
       this.linecolor = this.HsStylerColorService.findAndParseColor(
-        subStyle.getStroke().getColor()
+        subStyle.stroke.color
       );
     }
-    if (subStyle.getFill()?.getColor()) {
+    if (subStyle.stroke?.width) {
+      this.linewidth = subStyle.stroke.width;
+    }
+    if (subStyle.fill) {
       this.fillcolor = this.HsStylerColorService.findAndParseColor(
-        subStyle.getFill().getColor()
+        subStyle.fill
       );
     }
     if (subStyle.image) {
-      if (subStyle.image.getStroke()?.getColor()) {
+      this.getImageStyle(subStyle.image);
+    }
+  }
+  getImageStyle(image: any): void {
+    if (!image || image === undefined) {
+      return;
+    }
+    if (image.type === 'icon') {
+      if (
+        typeof image.src == 'string' &&
+        image.src.slice(0, 10) === 'data:image'
+      ) {
+        const encodedIconData = image.src.replace(
+          'data:image/svg+xml;base64,',
+          ''
+        );
+        this.serialized_icon = image.src;
+        const decodedIcon: any = this.decodeToUnicode(encodedIconData);
+        this.getIconStyleFromLayer(decodedIcon);
+      } else {
+        //TODO Check if this is even necessary
+        //this.iconSelected(image.getSrc());
+      }
+    }
+    if (image.type === 'circle') {
+      if (image.radius) {
+        this.radius = image.radius;
+      }
+      if (image.stroke?.width) {
+        this.iconlinewidth = image.stroke.width;
+      }
+      if (image.fill) {
+        this.iconfillcolor = this.HsStylerColorService.findAndParseColor(
+          image.fill
+        );
+      }
+      if (image.stroke?.color) {
         this.iconlinecolor = this.HsStylerColorService.findAndParseColor(
-          subStyle.getStroke().getColor()
+          image.stroke.color
         );
       }
-      if (subStyle.image.getFill()?.getColor()) {
+    }
+  }
+  getIconStyleFromLayer(decodedIcon: any): void {
```

Comment:
Would like to move these function as much as possible to service.

Reply:
Will do!